### PR TITLE
feat(haqq-website): sendgrid integration

### PR DESCRIPTION
**Pull request by hadzhehsen**

### CODE REVIEW

File: libs/website/forms/src/lib/subscribe-form/subscribe-form.tsx [hunk 1]
```diff
@@ -19,13 +19,13 @@ const schema: yup.ObjectSchema<SubscribeFormFields> = yup
   })
   .required();
 
-function submitForm(form: SubscribeFormFields): Promise<{ status: number }> {
-  return new Promise((resolve) => {
-    setTimeout(() => {
-      resolve({ status: 200 });
-    }, 2500);
-  });
-}
```
Comment (olegshilov) [inline libs/website/forms/src/lib/subscribe-form/subscribe-form.tsx]: send data to sendgrid from this method

File: libs/website/forms/src/lib/subscribe-form/subscribe-form.tsx [hunk 2]
```diff
@@ -42,22 +42,46 @@ export function SubscribeForm({
...
-        setSubscribeFormState(FormState.error);
-      }
+    try {
+      const response = await fetch('/api/sendgrid', request);
+      const data = await response.json();
+      return { status: response.status, data };
     } catch (error) {
       console.error(error);
-      setSubscribeFormState(FormState.error);
+      return { status: 500, data: error };
```
Comment (olegshilov) [inline libs/website/forms/src/lib/subscribe-form/subscribe-form.tsx]: revert all of this.
you need to just correctly handle response and error from `submitForm` method

File: libs/website/forms/src/lib/subscribe-form/subscribe-form.tsx (hunk 1 above)
Comment (olegshilov) [reply libs/website/forms/src/lib/subscribe-form/subscribe-form.tsx]: and use `axios.post()` instead fetch

**Review by olegshilov — CHANGES_REQUESTED** [review]

File: apps/website/pages/api/sendgrid.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import client from '@sendgrid/client';
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+client.setApiKey(process.env.SENDGRID_API_KEY);
```
Comment (olegshilov) [inline apps/website/pages/api/sendgrid.ts]:
```suggestion
client.setApiKey(process.env['SENDGRID_API_KEY']);
```

File: apps/website/pages/api/sendgrid.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+async function submitEmail(req: NextApiRequest, res: NextApiResponse) {
+  const data = {
+    list_ids: [process.env.SENDGRID_LIST_ID],
```
Comment (olegshilov) [inline apps/website/pages/api/sendgrid.ts]:
```suggestion
    list_ids: [process.env['SENDGRID_LIST_ID']],
```

**Review by olegshilov — CHANGES_REQUESTED** [review]

File: libs/website/forms/src/lib/subscribe-form/subscribe-form.tsx (hunk 1 above)
Comment (hadzhehsen) [reply libs/website/forms/src/lib/subscribe-form/subscribe-form.tsx]: Done

File: libs/website/forms/src/lib/subscribe-form/subscribe-form.tsx (hunk 2 above)
Comment (hadzhehsen) [reply libs/website/forms/src/lib/subscribe-form/subscribe-form.tsx]: Done

File: apps/website/pages/api/sendgrid.ts (hunk 3 above)
Comment (hadzhehsen) [reply apps/website/pages/api/sendgrid.ts]: Done

File: apps/website/pages/api/sendgrid.ts (hunk 4 above)
Comment (hadzhehsen) [reply apps/website/pages/api/sendgrid.ts]: Done

**Review by olegshilov — APPROVED** [review]

**Pull request merged**